extension/src/lambda_request_error.rs

```diff
@@ -64,53 +64,22 @@ impl From<&LambdaRequestError> for ExitReason {
 
 impl LambdaRequestError {
   pub fn is_fatal(&self) -> bool {
-    use LambdaRequestError::*;
-
-    let fatal_errors = [
-      // Add other fatal errors here
-      AppConnectionUnreachable,
-    ];
-
-    fatal_errors.contains(self)
-  }
-
-  pub fn worse(self, other: Self) -> Self {
-    use LambdaRequestError::*;
-
-    let ordered_reasons = [
-      RouterLambdaInvokeInvalid,
-      RouterUnreachable,
-      RouterConnectionError,
-      AppConnectionUnreachable,
-      AppConnectionError,
-      ChannelErrorOther,
-    ];
-
-    for reason in ordered_reasons {
-      if self == reason || other == reason {
-        return reason;
-      }
-    }
-
-    self // If no match is found, return the current value
+    // Add other fatal errors to the match pattern. For example:
+    // matches!(self, Self::Variant1 | Self::Variant2)
+    matches!(self, Self::AppConnectionUnreachable)
   }
+}
 
-  // This function will cause a compile-time error if a new variant is added to the enum
-  // but not added to the match expression.
-  #[allow(dead_code)]
-  fn ensure_all_variants_handled(variant: Self) {
-    use LambdaRequestError::*;
-
-    match variant {
-      // HEY - If you add here you need to add to the `worse` function array above
-      RouterLambdaInvokeInvalid => {}
-      RouterUnreachable => {}
-      RouterConnectionError => {}
-      AppConnectionUnreachable => {}
-      AppConnectionError => {}
-      ChannelErrorOther => {}
+impl Severity for LambdaRequestError {
+  fn severity(&self) -> usize {
+    match self {
+      Self::RouterLambdaInvokeInvalid => 0,
+      Self::RouterUnreachable => 1,
+      Self::RouterConnectionError => 2,
+      Self::AppConnectionUnreachable => 3,
+      Self::AppConnectionError => 4,
+      Self::ChannelErrorOther => 5,
```

Comment:
Since the test ensuring all members are added to the ordered array had to list all members in a match, we may as well explicitly assign them a severity. 

Another option would be to define the variants in order when defining the enum. Then you can cast the enum into a `usize` and compare them. But that also is easy to break by reordering the variant definitions or defining a new one in the wrong place.